wagtail_transfer/operations.py

```diff
@@ -428,6 +433,15 @@ def dependencies(self):
                         (model, id, 'exists')
                     )
 
+            elif isinstance(field, StreamField):
+                for model, id in get_object_references(field.stream_block, json.loads(self.object_data['fields'].get(field.name))):
+                    # TODO: add config check here
+                    if id == pk:
+                        continue
```

Comment:
Unless there's a specific gotcha I'm missing which causes a page's own ID to routinely appear in rich text / streamfield data, I don't think this check belongs here (and ditto for the corresponding line for the rich text case, which I missed earlier). While circular references will indeed probably throw a spanner in the works right now, they're something we'll ultimately have to deal with in the main dependency-handling logic, because the more general case (i.e. not just A linking to A, but A linking to B linking to A) can't be handled here by looking at the field in isolation. Filtering out references to self in this part of the code seems like sweeping the problem under the rug.

(If we _do_ keep this check in, it needs to check `model` as well, so that we're not erroneously skipping over references to image ID 123 on page ID 123.)

Reply:
Okay, that make sense - yep, I'll remove that

Reply:
Removed both in latest commit